src/components/IOUConfirmationList.js

```diff
@@ -328,84 +321,174 @@ class IOUConfirmationList extends Component {
-                        selectedOptions={this.getSelectedOptions()}
-                        onSelectRow={toggleOption}
-                        isDisabled={!this.props.isGroupSplit}
-                        optionHoveredStyle={hoverStyle}
-                    />
-                </ScrollView>
-                <View style={[styles.ph5, styles.pv5, styles.flexGrow1, styles.flexShrink0, styles.iouConfirmComment]}>
-                    <TextInput
-                        ref={el => this.textInput = el}
-                        label={this.props.translate('iOUConfirmationList.whatsItFor')}
-                        value={this.props.comment}
-                        onChangeText={this.props.onUpdateComment}
-                        placeholder={this.props.translate('common.optional')}
-                        placeholderTextColor={themeColors.placeholderText}
-                    />
-                </View>
-                <FixedFooter>
-                    {this.props.network.isOffline && (
-                        <Text style={[styles.formError, styles.pb2]}>
-                            {this.props.translate('session.offlineMessage')}
-                        </Text>
-                    )}
-                    {shouldShowSettlementButton ? (
-                        <SettlementButton
-                            isDisabled={shouldDisableButton}
-                            isLoading={this.props.iou.loading && !this.props.network.isOffline}
-                            onPress={this.onPress}
-                            shouldShowPaypal={Boolean(recipient.payPalMeAddress)}
-                            recipientPhoneNumber={recipient.phoneNumber}
-                            enablePaymentsRoute={ROUTES.IOU_SEND_ENABLE_PAYMENTS}
-                            addBankAccountRoute={ROUTES.IOU_SEND_ADD_BANK_ACCOUNT}
-                            addDebitCardRoute={ROUTES.IOU_SEND_ADD_DEBIT_CARD}
-                            currency={this.props.iou.selectedCurrencyCode}
-                        />
-                    ) : (
-                        <ButtonWithMenu
-                            isDisabled={shouldDisableButton}
-                            isLoading={isLoading}
-                            onPress={(_event, value) => this.onPress(value)}
-                            options={this.splitOrRequestOptions}
-                        />
-                    )}
-                </FixedFooter>
-            </>
+            <ArrowKeyFocusManager
+                initialFocusedIndex={this.allOptions.length}
+                listLength={this.allOptions.length + 1}
+                onFocusedIndexChanged={this.scrollToIndex}
+                onEnterKeyPressed={this.maybeToggleParticipant}
+                shouldEnterKeyEventBubble={focusedIndex => !this.props.hasMultipleParticipants || !this.allOptions[focusedIndex]}
+            >
+                {({focusedIndex}) => (
```

Comment:
**thought:** I have not tested yet but have a feeling rendering a function here might have perf implications. OptionsList can have lots and lots of things in it and can be slow to initialize. Have we tested the changes against production to see the effects with a large list of users?

Reply:
focusedIndex was lifted, so we aren't rendering a function here anymore 🙂